docs/reference/contributing/target/Watchdog.md

```diff
@@ -1,25 +1,45 @@
-<h3 id="watchdog-port">Watchdog</h3>
+### Watchdog
 
-[Include a brief description here.]
+The Watchdog HAL API provides a low-level interface to the Independent Watchdog Timer of a target. Implementing the Watchdog API is not mandatory, but by implementing it, you can use the hardware watchdog timer to detect and recover from computer malfunctions.
+
+Implement the Reset Reason API when implementing the Watchdog API. The Reset Reason API allows you to detect at boot time whether a watchdog caused the last reset.
 
 #### Assumptions
 
 ##### Defined behavior
 
-[Include any defined behavior in bullet format here.]
+- Sleep and debug modes don't stop the watchdog from counting down.
+- The function `hal_watchdog_init` is safe to call repeatedly. It must not do anything if you have already initialized the watchdog timer.
+- Maximum supported timeout is `UINT32_MAX` milliseconds; minimum timeout is 1ms.
+- The timeout must be accurate to the nearest millisecond.
 
 ##### Undefined behavior
 
-[Include any undefined behavior in bullet format here.]
+- Calling any function other than `hal_watchdog_init` or `hal_watchdog_get_platform_features` before you have initialized the watchdog.
 
 ##### Potential bugs
 
-[Include any potential bugs in bullet format here.]
+- A software reset may not stop the watchdog timer; the behavior is platform specific.
+- Timing on most platforms is based on the timeout registers and a prescaler value. They should be accurate to the nearest millisecond but may be off my a several µs.
+
+#### Dependency
+
+Hardware Independent Watchdog support.
 
 #### Implementing Watchdog
 
-[Include implementation information here.]
+You can find the API and specification for the Watchdog API in the following header file:
+
+[![View code](https://www.mbed.com/embed/?type=library)](https://github.com/ARMmbed/mbed-os/blob/feature-watchdog/hal/watchdog_api.h)
+
+To enable Watchdog support in Mbed OS, add the `WATCHDOG` label in the `device_has` option of the target's section in the `targets.json` file.
```

Comment:
Query: In this line, is "Watchdog" referring to the API or something else?

Reply:
Watchdog HAL API